staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```diff
@@ -286,7 +287,19 @@ func (o *AutoscaleOptions) createHorizontalPodAutoscaler(refName string, mapping
 	}
 	if o.CPUPercent >= 0 {
 		c := int32(o.CPUPercent)
-		scaler.Spec.TargetCPUUtilizationPercentage = &c
+
+		scaler.Spec.Metrics = []autoscalingv2.MetricSpec{
+			{
+				Type: autoscalingv2.ResourceMetricSourceType,
+				Resource: &autoscalingv2.ResourceMetricSource{
+					Name: "cpu-percent",
+					Target: autoscalingv2.MetricTarget{
+						Type:               autoscalingv2.UtilizationMetricType,
+						AverageUtilization: &c,
+					},
+				},
+			},
+		}
```

Comment:
can you please add links to relevant API docs here to help review this change, thanks!